Fix for 32890

prabindh opened the pull request:
Usage fix for https://github.com/tensorflow/tensorflow/issues/32890

prabindh committed:
Fix for 32890

Usage fix for https://github.com/tensorflow/tensorflow/issues/32890

prabindh commented:
@ichisadashioko The change is as per the TF2.0 moments API at - https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/nn/moments 
![image](https://user-images.githubusercontent.com/1215029/65828451-ab75e180-e2b8-11e9-9d48-02f8b5b16517.png)

ichisadashioko commented:
Do you know that there is [`r2.0`](https://github.com/tensorflow/tensorflow/blob/64c3d382cadf7bbe8e7e99884bede8284ff67f56/tensorflow/python/ops/nn_impl.py#L1180) branch.

At the time of writing, [`r2.0`](https://github.com/tensorflow/tensorflow/tree/r2.0/tensorflow) was 451 commits ahead, 6650 commits behind `master`.

prabindh commented:
Do we need to look at below instead for v2 ? https://github.com/tensorflow/tensorflow/blob/64c3d382cadf7bbe8e7e99884bede8284ff67f56/tensorflow/python/ops/nn_impl.py#L1245

mihaimaruseac commented on tensorflow/python/ops/nn_impl.py:
```diff
@@ -1394,7 +1394,7 @@ def batch_normalization(x,
       others which are being normalized over.
       `mean` and `variance` in this case would typically be the outputs of
-      `tf.nn.moments(..., keep_dims=True)` during training, or running averages
+      `tf.nn.moments(..., keepdims=True)` during training, or running averages
```
This is `keepdims` in `master`, thank you for fixing the documentation to have this correct.

https://github.com/tensorflow/tensorflow/blob/8569d12747c13ac16e864db1482b6754904371c8/tensorflow/python/ops/nn_impl.py#L1241

prabindh commented:
There is a set of errors in the CI like "found a whitelisted error:
  tensorflow/python/kernel_tests/constant_op_eager_test.py:250: [E0303(invalid-length-returned), ConstantTest.testInvalidLength.BadList.__len__] __len__ does not return non-negative integer".   These appear unrelated to the doc change. Do I need to retrigger any of these to get a pass ?

mihaimaruseac commented:
Whitelisted errors are ignored.

However, you do have other errors:

```
FAIL: Found 1 non-whitelisted pylint errors:
tensorflow/python/ops/nn_impl.py:1409: [C0301(line-too-long), ] Line too long (103/80)
```

prabindh committed:
Fix pylint error in PR 32893

Fix pylint error: tensorflow/python/ops/nn_impl.py:1409: [C0301(line-too-long), ] Line too long (103/80)
